_inc/client/state/modules/reducer.js

```diff
@@ -29,6 +30,19 @@ import {
 export const items = ( state = {}, action ) => {
 	switch ( action.type ) {
 		case JETPACK_SET_INITIAL_STATE:
+			Object.keys( action.initialState.getModules || {} ).map( function( slug ) {
```

Comment:
I would vote for this logic to be moved inside components somehow. Maybe [here](https://github.com/Automattic/jetpack/blob/master/_inc/client/engagement/index.jsx#L178) on each settings tab. That is, decoding in the `connect()` statements for each tab.

As it is now, this code makes this reducer depend on a global `document` variable and can't be tested as a pure function.

In fact, as a quick alternative, this could be moved to `admin.js`  before the call to `render()` and have it direc